core/src/main/java/org/pentaho/di/connections/vfs/VFSConnectionDetails.java

```diff
@@ -69,5 +71,14 @@ default String getRootPath() {
    *
    * @param rootPath The root path
    */
-  default void setRootPath(String rootPath ) {}
+  default void setRootPath( String rootPath ) { }
+
+  /**
+   * Gets the list of roles with access to the connection from the Business Analytics product.
+   * <p>
+   * Access control does not distinguish between types of access operations (such as read, write, delete).
+   * Access is granted to either all or none of the operations.
+   * @return A non-null list of roles.
+   */
+  List<String> getBaRoles();
```

Comment:
not adding a default here also means the build will be tricky. You'll need to land the kettle changes and the downstream repo changes in the same day.